### PR TITLE
carry #3126: linux: Support setting execution domain via linux personality

### PR DESCRIPTION
**Pull request by Zheaoli**
carry #3126

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: libcontainer/specconv/spec_linux.go [hunk 1]
```diff
@@ -573,6 +582,17 @@ func checkPropertyName(s string) error {
 	return nil
 }
 
+// getLinuxPersonalityFromStr converts the string domain received from spec to equivalent integer.
+func getLinuxPersonalityFromStr(domain string) (int, error) {
+	// defaults to PerLinux
```
Comment (lifubang) [inline libcontainer/specconv/spec_linux.go:587]: What’s this comment mean?

Comment (Zheaoli) [reply libcontainer/specconv/spec_linux.go:587]: Useless comment, I will update it

**Comment by Zheaoli** [comment]
Just another problem, should we support personality flag in this PR? maybe it is a little bit difficult to test

File: libcontainer/specconv/spec_linux.go [hunk 2]
```diff
@@ -434,6 +434,18 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 				MemBwSchema:   spec.Linux.IntelRdt.MemBwSchema,
 			}
 		}
+		if spec.Linux.Personality != nil {
+			if spec.Linux.Personality.Flags != nil {
```
Comment (lifubang) [inline libcontainer/specconv/spec_linux.go:438]: How about ‘len(spec.Linux.Personality.Flags)>0’?

**Review by lifubang — DISMISSED** [review]

File: libcontainer/system/linux.go [hunk 3]
```diff
@@ -214,3 +214,13 @@ func Copy(dst io.Writer, src io.Reader) (copied int64, err error) {
 fallback:
 	return io.Copy(dst, src)
 }
+
+// SetLinuxPersonality sets the Linux execution personality. For more information see the personality syscall documentation.
+// checkout getLinuxPersonalityFromStr() from libcontainer/specconv/spec_linux.go for type conversion.
+func SetLinuxPersonality(personality int) error {
+	_, _, errno := syscall.Syscall(syscall.SYS_PERSONALITY, uintptr(personality), 0, 0)
```
Comment (lifubang) [inline libcontainer/system/linux.go:221]: How about ‘unix.Syscall’ and ‘unix.SYS_PERSONALITY’?

**Review by lifubang — CHANGES_REQUESTED** [review]
Sorry, forgot to request this change.

File: libcontainer/system/linux.go (hunk 3 above)
Comment (Zheaoli) [reply libcontainer/system/linux.go:221]: emm,let me think, means keep the same conversion style with other syscall?

Comment (lifubang) [reply libcontainer/system/linux.go:221]: https://pkg.go.dev/syscall
Deprecated: this package is locked down. Callers should use the corresponding package in the golang.org/x/sys repository instead. That is also where updates required by new systems or versions should be applied. See https://golang.org/s/go1.4-syscall for more information.

Comment (Zheaoli) [reply libcontainer/system/linux.go:221]: SGTM

**Review by lifubang — APPROVED** [review]

**Comment by kolyshkin** [comment]
This went into runc 1.2.0-rc.1

**Pull request merged**